torch/_inductor/cudagraph_trees.py

```diff
@@ -950,21 +980,23 @@ def __init__(
 
         self.graph.replay()
 
-    def _copy_inputs_and_remove_from_src(self, dsts, srcs):
+    def _copy_inputs_and_remove_from_src(
+        self, dsts: List[InputType], srcs: List[InputType]
+    ) -> None:
         dst_tensors = []
         src_tensors = []
         for idx in self.non_static_input_idx:
             if not isinstance(srcs[idx], torch.Tensor):
                 continue
             expanded_dims = self.expanded_dims[idx]
-            dst_tensors.append(index_expanded_dims(dsts[idx], expanded_dims))
-            src_tensors.append(index_expanded_dims(srcs[idx], expanded_dims))
-            srcs[idx] = None
+            dst_tensors.append(index_expanded_dims(dsts[idx], expanded_dims))  # type: ignore[arg-type]
+            src_tensors.append(index_expanded_dims(srcs[idx], expanded_dims))  # type: ignore[arg-type]
+            srcs[idx] = None  # type: ignore[call-overload]
         # Fails on empty lists
         if dst_tensors:
             torch._foreach_copy_(dst_tensors, src_tensors)
 
-    def check_static_inputs_are_stable(self, new_inputs):
+    def check_static_inputs_are_stable(self, new_inputs: List[InputType]) -> None:
```

Comment:
Hey @BoyuanFeng, I think, this has broken mypy at the line 1004 below, where `List[Tensor]` is expected. I'm getting this when running `lintrunner -a --all-files --take MYPY` locally:

```
>>> Lint for torch/_inductor/cudagraph_trees.py:

  Error (MYPY) [arg-type]
    Argument 1 to "_tensors_data_ptrs_at_indices_equal" has incompatible type
    "list[Tensor | int]"; expected "list[Tensor]"

        1004  |        if (
        1005  |            not self.rerecord_if_static_inputs_change
        1006  |            and not torch._C._tensors_data_ptrs_at_indices_equal(
    >>> 1007  |                new_inputs,
        1008  |                self.static_input_data_ptrs,
        1009  |                self.non_managed_static_input_idxs,
        1010  |            )

  Error (MYPY) [arg-type]
    Argument 1 to "_tensors_data_ptrs_at_indices_equal" has incompatible type
    "list[Tensor | int]"; expected "list[Tensor]"

        1623  |        # this is on the hot path so moved to C++. equivalent to:
        1624  |        # return all(t.data_ptr() == data_ptr for (t, data_ptr) in zip(tensors, data_ptrs))
        1625  |        if not torch._C._tensors_data_ptrs_at_indices_equal(
    >>> 1626  |            inputs, self.static_input_data_ptrs, self.cudagraph_managed_idxs
        1627  |        ):
        1628  |            status = CheckInvariantStatus.CudagraphManagedIdxMismatch
        1629  |            _logger = functools.partial(

  Error (MYPY) [arg-type]
    Argument 1 to "_tensors_data_ptrs_at_indices_equal" has incompatible type
    "list[Tensor | int]"; expected "list[Tensor]"

        1646  |        if (
        1647  |            self.rerecord_if_static_inputs_change
        1648  |            and not torch._C._tensors_data_ptrs_at_indices_equal(
    >>> 1649  |                inputs, self.static_input_data_ptrs, self.static_input_idxs
        1650  |            )
        1651  |        ):
        1652  |            status = CheckInvariantStatus.StaticInputIdxMismatch
```

cc @eellison.

Reply:
Hmm, nevermind. I see that you've already addressed this. Pulling and rebuilding has fixed the mypy error. Please ignore the above; sorry for the noise.